File: php/features.php
<?php
declare(strict_types=1);

// Gutendex SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class GutendexFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new GutendexBaseFeature();
            case "test":
                return new GutendexTestFeature();
            default:
                return new GutendexBaseFeature();
        }
    }
}

File: py/features.py
# Gutendex SDK feature factory

from feature.base_feature import GutendexBaseFeature
from feature.test_feature import GutendexTestFeature


def _make_feature(name):
    features = {
        "base": lambda: GutendexBaseFeature(),
        "test": lambda: GutendexTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
